libbindgen/src/ir/context.rs

```diff
@@ -938,11 +938,9 @@ impl<'ctx> BindgenContext<'ctx> {
                     return true;
                 }
 
-                let name = item.canonical_name(self);
+                let name = item.real_canonical_name(self, false, true);
                 match *item.kind() {
-                    ItemKind::Module(..) => {
-                        self.options().enable_cxx_namespaces
-                    }
+                    ItemKind::Module(..) => true,
```

Comment:
Probably? I'll add a test.

Sounds like too complex given we don't have any real benefit? Alternatively, we can just generate the namespace, and if we didn't generate anything from it don't push the result, which seems slightly preferrable.